node/src/service.rs

```diff
@@ -225,15 +231,16 @@ async fn start_dev_impl(
         .flatten()
         .expect("Genesis block exists; qed");
 
-    if let Some(provider_options) = provider_options {
-        start_sh_services(
-            provider_options,
-            &task_manager,
-            genesis_hash,
-            &config,
-            &mut net_config,
-        )
-        .await;
+    // Spawning File Transfer Service if node is running as a Storage Provider.
+    // This is done here because the File Transfer Service modifies the network configuration.
+    let mut file_transfer_service_handle = None;
+    if provider_options.is_some() {
+        let task_spawner = TaskSpawner::new(task_manager.spawn_handle(), "generic");
+
+        file_transfer_service_handle = Some(
+            spawn_file_transfer_service(&task_spawner, genesis_hash, &config, &mut net_config)
+                .await,
+        );
     }
```

Comment:
Just out of curiosity: if we shadow the `file_transfer_service_handle` variable with its content when we spawn the Blockchain service (and we never use it for anything else beforehand), why is it needed to initiate it as a `Some()` value and then `.expect` in the Blockchain service initialization?